lib/ex_image_info/types/isobmff.ex

```diff
@@ -0,0 +1,313 @@
+
+      ispe_box =
+        state.ispe_boxes
+        |> Enum.reverse()
+        |> Enum.find(fn {idx, _} -> idx in primary_indices end)
+
+      case ispe_box do
+        {_idx, {width, height}} ->
+          if state.rotation in [90, 270] do
+            {height, width}
+          else
+            {width, height}
+          end
+
+        _ ->
+          # wrong primary_box assignation
+          @malformed
+      end
+    end
+  end
+
+  defp handle_box("hdlr", size, <<rest::binary>>, state) when size >= 12 do
+    case rest do
+      <<_::binary-size(8), "pict", _::binary-size(size - 8 - 4), rest::binary>> ->
+        state =
+          state
+          |> State.update_readable_bytes(-size)
+          |> State.update_index()
+
+        parse_box(rest, state)
+
+      <<_::binary-size(size), _rest::binary>> ->
+        @unimplemented
+
+      _ ->
+        @malformed
+    end
+  end
+
+  defp handle_box("hdlr", _size, <<_rest::binary>>, _state), do: @unimplemented
+
+  defp handle_box("pitm", size, <<rest::binary>>, state) do
+    case rest do
+      <<_::binary-size(4), pbox::size(16), _::binary-size(size - 4 - 2), rest::binary>> ->
+        state =
+          %{state | primary_box: pbox}
+          |> State.update_readable_bytes(-size)
+          |> State.update_index()
+
+        parse_box(rest, state)
+
+      _ ->
+        @malformed
+    end
+  end
+
+  # box types found in all images tested:
+  #
+  #     @skippable [
+  #       "iloc",
+  #       "iinf",
+  #       "hvcC",
+  #       "clap",
+  #       "pixi",
+  #       "iref",
+  #       "dinf",
+  #       "colr",
+  #       "auxC",
+  #       "mdat",
+  #       # Mac Preview App - rotate - export as heic generates this box:
+  #       "clli",
+  #       # avif
+  #       "pasp",
+  #       "av1C"
+  #     ]
+  #     defp handle_box(skippable, size, <<rest::binary>>, state)
+  #          when skippable in @skippable do
+  #       case rest do
+  #         <<_::binary-size(size), rest::binary>> ->
+  #           state = state |> State.update_readable_bytes(-size) |> State.update_index()
+  #           parse_box(rest, state)
+  #
+  #         _ ->
+  #           @malformed
+  #       end
+  #     end
+  #
+  # to increase robustness, changing the spec to accept any type, moving to the bottom of `handle_box` functions
+
+  @nestable ["iprp", "ipco"]
+  defp handle_box(nestable, _size, <<rest::binary>>, state)
+       when nestable in @nestable do
+    parse_box(rest, State.update_index(state, 0))
+  end
+
+  defp handle_box("ispe", size, <<rest::binary>>, state) when size >= 12 do
+    case rest do
+      <<_::binary-size(4), width::size(32), height::size(32), _::binary-size(size - 12),
+        rest::binary>> ->
+        state =
+          state
+          |> State.update_readable_bytes(-size)
+          |> State.ispe_box(width, height)
+          |> State.update_index()
+
+        parse_box(rest, state)
+
+      <<_::binary-size(size), _rest::binary>> ->
+        @unimplemented
```

Comment:
Future work/Contributions welcomed: no real example reaching this case (`ispe` box without width/height).